File: MuseoPOO/cuadro.py
"""Módulo que define la clase Cuadro."""

from datetime import date

from obra_de_arte import ObraDeArte


class Cuadro(ObraDeArte):
    """Representa un cuadro del museo."""

    def __init__(
        self,
        id_obra: int,
        titulo: str,
        autor: str,
        periodo: str,
        valoracion: float,
        fecha_creacion: date,
        fecha_entrada: date,
        estilo: str,
        tecnica: str,
        estado: str = "expuesta",
    ) -> None:
        super().__init__(
            id_obra, titulo, autor, periodo,
            valoracion, fecha_creacion, fecha_entrada, estado,
        )
        self._estilo = estilo
        self._tecnica = tecnica

    def get_estilo(self) -> str:
        return self._estilo

    def get_tecnica(self) -> str:
        return self._tecnica

    def consultar_info(self) -> str:
        return (
            f"Cuadro: '{self._titulo}' | Autor: {self._autor} | "
            f"Periodo: {self._periodo} | Estilo: {self._estilo} | "
            f"Técnica: {self._tecnica} | "
            f"Valoración: ${self._valoracion:,.2f} | "
            f"Estado: {self._estado}"
        )

File: Museo Java/Cuadro.java
import java.time.LocalDate;

public class Cuadro extends ObraDeArte {

    private String estilo;
    private String tecnica;

    public Cuadro(int idObra, String titulo, String autor, String periodo,
                  double valoracion, LocalDate fechaCreacion, LocalDate fechaEntrada,
                  String estilo, String tecnica, String estado) {
        super(idObra, titulo, autor, periodo, valoracion, fechaCreacion, fechaEntrada, estado);
        this.estilo = estilo;
        this.tecnica = tecnica;
    }

    public Cuadro(int idObra, String titulo, String autor, String periodo,
                  double valoracion, LocalDate fechaCreacion, LocalDate fechaEntrada,
                  String estilo, String tecnica) {
        this(idObra, titulo, autor, periodo, valoracion, fechaCreacion, fechaEntrada, estilo, tecnica, "expuesta");
    }

    public String getEstilo() { return estilo; }
    public String getTecnica() { return tecnica; }

    @Override
    public String consultarInfo() {
        return String.format("Cuadro: '%s' | Autor: %s | Periodo: %s | Estilo: %s | Técnica: %s | Valoración: $%,.2f | Estado: %s",
                titulo, autor, periodo, estilo, tecnica, valoracion, estado);
    }
}
